website/docs/plugin/framework/resources/configure.mdx

```diff
@@ -51,6 +53,8 @@ func (p *ExampleCloudProvider) Configure(ctx context.Context, req provider.Confi
 
 Implement the [`resource.ResourceWithConfigure` interface](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithConfigure) which receives the provider configured data from the [`Provider` interface `Configure` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/provider#Provider.Configure) and saves it into the [`resource.Resource` interface](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#Resource) implementation.
 
+The [`resource.ResourceWithConfigure` interface `Configure` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithConfigure.Configure) is called during execution of the [`terraform validate`](/cli/commands/validate), [`terraform plan`](/cli/commands/plan) and [`terraform apply`](/cli/commands/apply) commands when the [`ValidateResourceConfig`](/plugin/framework/internals/rpcs#validateresourceconfig-rpc) RPC is sent. Additionally, the [`resource.ResourceWithConfigure` interface `Configure` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithConfigure.Configure) is called during execution of the [`terraform plan`](/cli/commands/plan) and [`terraform apply`](/cli/commands/apply) commands when the [`ReadResource`](/plugin/framework/internals/rpcs#readresource-rpc) RPC is sent.
```

Comment:
Similarly --

I think if we are going to call out this level of detail, then we should clarify that Terraform calling the `ValidateDataResourceConfig` RPC would _not_ call the `ConfigureProvider` RPC first, so implementations need to account for that situation. Configuration validation should be "offline" in Terraform.